src/participant/participant.ts

```diff
@@ -590,6 +615,42 @@ export default class ParticipantController {
     return true;
   }
 
+  _shouldFetchCollectionSchema(): boolean {
```

Comment:
Hmmm this method modifies the state but the signature looks harmless. I would use a name that makes explicit there are side effects or changing the state in another method.